pkg/connectors/clients/policymanager_grpc.go

```diff
@@ -39,11 +46,333 @@ func NewGrpcPolicyManager(name string, connectionURL string, connectionTimeout t
+		operation := datasets[i].GetOperation()
+		destination := operation.GetDestination()
+		action.SetDestination(destination)
+		operationType := operation.GetType()
+		if operationType == pb.AccessOperation_READ {
+			action.SetActionType(openapiclientmodels.READ)
+		}
+		if operationType == pb.AccessOperation_WRITE {
+			action.SetActionType(openapiclientmodels.WRITE)
+		}
+		datasetID := datasets[i].GetDataset().GetDatasetId()
+		resource.SetName(datasetID)
+	}
+	req.SetResource(resource)
+
+	processingGeo := in.GetAppInfo().GetProcessingGeography()
+	action.SetProcessingLocation(processingGeo)
+	req.SetAction(action)
+
+	reqContext := make(map[string]interface{})
+	properties := in.GetAppInfo().GetProperties()
+	reqContext["intent"] = properties["intent"]
+	reqContext["role"] = properties["role"]
+	req.SetContext(reqContext)
+
+	return &req, creds, nil
+}
+
+func ConvertOpenAPIReqToGrpcReq(in *openapiclientmodels.PolicyManagerRequest, creds string) (*pb.ApplicationContext, error) {
+	credentialPath := creds
+	action := in.GetAction()
+	processingGeo := (&action).GetProcessingLocation()
+	log.Println("processingGeo: ", processingGeo)
+
+	properties := make(map[string]string)
+	context := in.GetContext()
+	if intent, ok := context["intent"].(string); ok {
+		properties["intent"] = intent
+	}
+	if role, ok := context["role"].(string); ok {
+		properties["role"] = role
+	}
+
+	appInfo := &pb.ApplicationDetails{ProcessingGeography: processingGeo, Properties: properties}
+
+	datasetContextList := []*pb.DatasetContext{}
+	resource := in.GetResource()
+	datasetID := (&resource).GetName()
+	dataset := &pb.DatasetIdentifier{DatasetId: datasetID}
+
+	destination := (&action).GetDestination()
+	actionType := (&action).GetActionType()
+
+	var grpcActionType pb.AccessOperation_AccessType
+	switch actionType {
+	case openapiclientmodels.READ:
+		grpcActionType = pb.AccessOperation_READ
+	case openapiclientmodels.WRITE:
+		grpcActionType = pb.AccessOperation_WRITE
+	default: // default is read
+		grpcActionType = pb.AccessOperation_READ
+	}
+
+	operation := &pb.AccessOperation{Type: grpcActionType, Destination: destination}
+	datasetContext := &pb.DatasetContext{Dataset: dataset, Operation: operation}
+	datasetContextList = append(datasetContextList, datasetContext)
+
+	appContext := &pb.ApplicationContext{CredentialPath: credentialPath, AppInfo: appInfo, Datasets: datasetContextList}
+
+	log.Println("Constructed GRPC appContext: ", appContext)
+
+	return appContext, nil
+}
+
+func ConvertOpenAPIRespToGrpcResp(
+	out *openapiclientmodels.PolicyManagerResponse,
+	datasetID string, op *pb.AccessOperation) (*pb.PoliciesDecisions, error) {
+	res, err := json.MarshalIndent(out, "", "\t")
+	log.Println("err :", err)
+	log.Println("Marshalled response in ConvertOpenAPIRespToGrpcResp:", string(res))
+
+	resultItems := out.GetResult()
+	enforcementActions := make([]*pb.EnforcementAction, 0)
+	usedPolicies := make([]*pb.Policy, 0)
```

Comment:
is these policies or policy decisions? should we clarify if they are decisions?

Reply:
usedPolicies are policies which when evaluated led to the corresponding enforcement actions.